passes/techmap/flatten.cc

```diff
@@ -220,7 +221,26 @@ struct FlattenWorker
 			sigmap.add(new_conn.first, new_conn.second);
 		}
 
+		RTLIL::Cell *scopeinfo = nullptr;
+		RTLIL::IdString cell_name = cell->name;
+
+		if (create_scopeinfo)
+		{
+			// The $scopeinfo's name will be changed below after removing the flattened cell
+			scopeinfo = module->addCell(NEW_ID, ID($scopeinfo));
+			scopeinfo->setParam(ID::TYPE, RTLIL::Const("cell"));
```

Comment:
You have probably thought this out already and I am curious to hear it. Should this be `cell` or `module`?

Reply:
(Also sorry for bikeshedding but the only other occurences of enums on cell parameters within Yosys I have seen were on some vendor cells, where the values were fully uppercase. Also e.g. in select syntax we would have the parameter name and its value next to each other. For reasons of consistence we could say the values for an enum-style parameter should be uppercase.)

Reply:
> You have probably thought this out already and I am curious to hear it. Should this be cell or module?

It refers to a module instantiation (a `module` was written by the designer at some point) so I would say "module" and not "cell".

> (Also sorry for bikeshedding but the only other occurences of enums on cell parameters within Yosys I have seen were on some vendor cells, where the values were fully uppercase. Also e.g. in select syntax we would have the parameter name and its value next to each other. For reasons of consistence we could say the values for an enum-style parameter should be uppercase.)

I think it's consistent enough with lowercase. In Verilog and RTLIL both the module is spelled as `module`, and when extending this I think we might have other scopes such as `begin` or maybe `struct`. We aren't writing ALGOL to have `BEGIN` :)

Reply:
> I think it's consistent enough with lowercase. In Verilog and RTLIL both the module is spelled as module, and when extending this I think we might have other scopes such as begin or maybe struct. We aren't writing ALGOL to have BEGIN :)

No contest

Reply:
I couldn't decide between "cell" and "module" as which one is more appropriate depends on whether you want to indicate what caused the presence of the scope i.e. the entity that has the same hierarchical name as the `$scopeinfo` cell, in which case it would be "cell" (or "instance" if we're using verilog not RTLIL terminology). If we're indicating what defines the contents of the scope then "module" is clearly the correct choice.

Since both approaches make sense to me and changing it is trivial at this point, I went with an arbitrary choice.

Given the feedback so far, I'd change this to "module".

Reply:
I am not opposed to `instance`, though I feel it might be a little too ambiguous.